ultralytics/yolo/engine/exporter.py

```diff
@@ -525,18 +525,42 @@ def _export_saved_model(self, prefix=colorstr('TensorFlow SavedModel:')):
         f_onnx, _ = self._export_onnx()
 
         # Export to TF
-        int8 = '-oiqt -qt per-tensor' if self.args.int8 else ''
+        if self.args.int8:
+            import numpy as np
+
+            from ultralytics.yolo.data.dataloaders.stream_loaders import LoadImages
+            from ultralytics.yolo.data.utils import check_det_dataset
+
+            # Generate calibration data for integer quantization
+            _, _, *imgsz = list(self.im.shape)  # BCHW
+            dataset = LoadImages(check_det_dataset(self.args.data)['train'], imgsz=imgsz, auto=False)
+            calib_data = []
+            n_images = 100
+            for n, (_, img, _, _, _) in enumerate(dataset):
+                if n >= n_images:
+                    break
+                img = img.transpose(1, 2, 0)  # CHW to HWC
+                img = img[np.newaxis]
+                calib_data.append(img)
+            f.mkdir()
+            calib_file = Path(f / 'calib_data.npy')
+            np.save(calib_file, np.vstack(calib_data))  # BHWC
+
+            int8 = f'-oiqt -qt per-tensor -qcind images {calib_file} "[[[[0, 0, 0]]]]" "[[[[255, 255, 255]]]]"'
```

Comment:
Is there any particular reason to use per-tensor quantization (`-qt per-tensor` option)?
If this option is dropped, per-channel quantization is applied to weight tensors and leads to further accuracy improvements.

preprocess calib (2) | normalize pred xywh (1) | per-channel quant | mAP50 | mAP50-95
-- | -- | -- | -- | --
x | x |   | 54.5 | 36.8
x | x | x | 55.8 | 38.6

Reply:
I also tested the integer model on COCO2017 val, and found mAP difference was much smaller between per-tensor and per-channel models. So I think we can keep `-qt per-tensor` option here.